src/probnum/diffeq/odefiltsmooth/odefiltsmooth.py

```diff
@@ -218,196 +254,187 @@ def probsolve_ivp(
+        # lazy version of Hairer, Wanner, Norsett, p. 169
+        norm_y0 = np.linalg.norm(ivp.initrv.mean)
+        norm_dy0 = np.linalg.norm(ivp(ivp.t0, ivp.initrv.mean))
+        firststep = 0.01 * norm_y0 / norm_dy0
+    else:
+        firststep = step
+
+    # Set up some default hyperparameters: not great, but much better than 1.0
+    lengthscale = lengthscale if lengthscale is not None else 1.0 / firststep
+    driftspeed = driftspeed if driftspeed is not None else firststep
+
+    # Construct underlying Kalman filter object
+    stprl = _create_steprule(atol, rtol, step, firststep)
+    prior = _string2prior(ivp, which_prior, driftspeed, lengthscale)
+    gfilt = _create_filter(ivp, prior, method)
+
+    # Solve
+    solver = GaussianIVPFilter(ivp, gfilt, with_smoothing=dense_output)
     solution = solver.solve(steprule=stprl)
     return solution
 
 
-def _create_filter(ivp, prior, method, **kwargs):
+def _create_filter(ivp, prior, method):
     """Create the solver object that is used."""
-    if method not in ["ekf0", "ekf1", "ukf", "eks0", "eks1", "uks"]:
-        raise ValueError("Method not supported.")
-    gfilt = _string2filter(ivp, prior, method, **kwargs)
+    if method not in ["EK0", "EK1", "UK"]:
+        raise ValueError("This method is not supported.")
+    gfilt = _string2filter(ivp, prior, method)
     return gfilt
 
 
-def _create_steprule(atol, rtol, step, firststep, ivp):
-    _check_step_tol(step, atol, rtol)
+def _create_steprule(atol, rtol, step, firststep):
+    if atol is None and rtol is None and step is None:
+        errormsg = (
+            "Please specify either absolute and relative tolerances or a step size."
+        )
+        raise ValueError(errormsg)
 
-    if step is not None:
+    if atol is None and rtol is None:
         stprl = steprule.ConstantSteps(step)
     else:
-        if firststep is None:
-            # lazy version of Hairer, Wanner, Norsett, p. 169
-            norm_y0 = np.linalg.norm(ivp.initrv.mean)
-            norm_dy0 = np.linalg.norm(ivp(ivp.t0, ivp.initrv.mean))
-            firststep = 0.01 * norm_y0 / norm_dy0
+        firststep = step if step is not None else firststep
         stprl = steprule.AdaptiveSteps(firststep=firststep, atol=atol, rtol=rtol)
     return stprl
 
 
-def _check_step_tol(step, atol, rtol):
-    both_none = atol is None and rtol is None and step is None
-    both_not_none = (atol is not None and rtol is not None) and step is not None
-    if both_none or both_not_none:
-        errormsg = "Please specify either a tolerance or a step size."
-        raise ValueError(errormsg)
-    atol_not_rtol = atol is not None and rtol is None
-    rtol_not_atol = rtol is not None and atol is None
-    if atol_not_rtol or rtol_not_atol:
-        errormsg = "Please specify either both atol and rtol, or neither."
-        raise ValueError(errormsg)
-
-
-def _string2prior(ivp, which_prior, **kwargs):
+def _string2prior(ivp, which_prior, driftspeed, lengthscale):
```

Comment:
Alright, so I tried something:
```python
import re
which_prior = "IOUP14"
m = re.match("^(IBM|IOUP|MAT)([0-9]+)$", which_prior)
if m is None:
    raise ValueError("This prior is not supported.")
prior_str, order_str = m.groups()
```
after which we have `prior_str == "IOUP"` and `order_str == "14"`. So all that's left would be an if-else / dict matching over the prior, and some special care for the matern order.

What do you think?

Reply:
Nice, thanks! Tbh, this looks better than I expected^^ I played around with it, and like its robustness!